src/Symfony/Component/EventDispatcher/Debug/WrappedListener.php

```diff
@@ -46,7 +46,13 @@ public function __construct($listener, $name, Stopwatch $stopwatch, EventDispatc
             $this->name = is_object($listener[0]) ? get_class($listener[0]) : $listener[0];
             $this->pretty = $this->name.'::'.$listener[1];
         } elseif ($listener instanceof \Closure) {
-            $this->pretty = $this->name = 'closure';
+            $r = new \ReflectionFunction($listener);
+            if (preg_match('#^/\*\* @closure-proxy ([^: ]++)::([^: ]++) \*/$#', $r->getDocComment(), $m)) {
```

Comment:
hehe, I was wondering why you had this phpdoc comment in the container generation :) Note that this trick does only work with a dumped container. Not a big deal though.